lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -776,12 +782,19 @@ class DocumentImpl extends NodeImpl {
   }
 
   createNodeIterator(root, whatToShow, filter) {
-    const nodeIterator = NodeIterator.createImpl(this._globalObject, [], { root, whatToShow, filter });
+    const nodeIterator = NodeIterator.createImpl(this._globalObject, [], {
+      root, whatToShow, filter,
+      workingNodeIteratorsMax: hasWeakRefs ? null : this._workingNodeIteratorsMax
```

Comment:
I don't think we need to check hasWeakRefs here since it's checked in NodeIterator-impl. (Alternately, you could remove the checks there, and check if `_workingNodeIteratorsMax` is null.)